zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/AuthorizationCheckBehavior.java

```diff
@@ -216,16 +214,15 @@ public List<String> getAuthorizedTenantIds(final TypedRecord<?> command) {
     if (command.getAuthorizations().get(Authorization.AUTHORIZED_TENANTS) != null) {
       return (List<String>) command.getAuthorizations().get(Authorization.AUTHORIZED_TENANTS);
     }
-    final var userKey = getUserKey(command);
     final List<String> authorizedTenantIds = new ArrayList<>();
     authorizedTenantIds.add(TenantOwned.DEFAULT_TENANT_IDENTIFIER);
-    return userKey
+    return getUser(command)
         .map(
-            key ->
-                userState
-                    .getUser(key)
-                    .map(PersistedUser::getTenantIdsList)
-                    .orElse(authorizedTenantIds))
+            user -> {
+              final List<String> tenantIds = user.getTenantIdsList();
+              // Use default tenant IDs if user's tenant list is empty
+              return tenantIds.isEmpty() ? authorizedTenantIds : tenantIds;
```

Comment:
💭 I'm not sure about this. This bug came up while testing the resolve incident gRPC endpoint. The incident record by default has the default tenant, I'm assuming in this scenario if the user doesn't have any tenant it is ok to return the default one